crates/sargon/src/system/sargon_os/sargon_os_security_structures.rs

```diff
@@ -100,6 +100,35 @@ impl SargonOS {
     }
 }
 
+impl SargonOS {
+    /// Returns the status of the prerequisites for building a Security Shield.
+    ///
+    /// According to [definition][doc], a Security Shield can be built if the user has, asides from
+    /// the Identity factor, "2 or more factors, one of which must be Hardware"
+    ///
+    /// [doc]: https://radixdlt.atlassian.net/wiki/spaces/AT/pages/3758063620/MFA+Rules+for+Factors+and+Security+Shields#Factor-Prerequisites
+    pub fn get_security_shield_prerequisites_status(
+        &self,
+    ) -> Result<SecurityShieldPrerequisitesStatus> {
+        let factor_sources = self.factor_sources()?;
```

Comment:
I think we should try to have a thin functions on SargonOs as possible. So i would put this on Profile. And go through ProfileStateHolder